shaman/controllers/builds/projects.py

```diff
@@ -0,0 +1,73 @@
+from pecan import expose, abort, request
+from pecan.secure import secure
+
+from shaman.models import Project, Build, Arch
+from shaman.controllers.repos import refs
+from shaman.auth import basic_auth
+from shaman import models
+
+
+class ProjectAPIController(object):
+
+    def __init__(self, project_name):
+        self.project_name = project_name
+        self.project = Project.query.filter_by(name=project_name).first()
+        if not self.project:
+            if request.method != 'POST':
+                abort(404)
+        else:
+            request.context['project_id'] = self.project.id
+
+    @expose(generic=True, template='json')
+    def index(self):
+        abort(405)
+
+    @index.when(method='GET', template='json')
+    def index_get(self):
+        return list(
+            set([r.ref for r in self.project.repos])
+        )
+
+    #TODO: we need schema validation on this method
+    @secure(basic_auth)
+    @index.when(method='POST', template='json')
+    def index_post(self):
+        if not self.project:
+            self.project = models.get_or_create(Project, name=self.project_name)
+        chacra_url = request.json["chacra_url"]
+        repo = Build.query.filter_by(chacra_url=chacra_url).first()
+        if not repo:
+            data = dict(
+                project=self.project,
+                ref=request.json["ref"],
+                sha1=request.json["sha1"],
+                flavor=request.json["flavor"],
+                distro=request.json["distro"],
```

Comment:
``distro``, ``distro_version`` and ``arch`` are no longer part of the ``Build`` model